### PR TITLE
[Fix](count on index) fix count on index opt when count project expr

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

This PR addresses an issue where counting on an index with a projected expression was suboptimal.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java [hunk 1]
```diff
@@ -541,26 +541,58 @@ private LogicalAggregate<? extends Plan> pushdownCountOnIndex(
...
-                                            physicalOlapScan,
-                                            PushDownAggOp.COUNT_ON_MATCH)))))
-            ));
-        } else {
-            return agg.withChildren(ImmutableList.of(
-                            filter.withChildren(ImmutableList.of(
-                                    new PhysicalStorageLayerAggregate(
-                                            physicalOlapScan,
-                                            PushDownAggOp.COUNT_ON_MATCH)))));
+            arguments = Project.findProject(arguments, project.getProjects())
```
Comment (csun5285) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:575]: findProject will throw an exception, so a try-catch block is needed here

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**